shark/shark_downloader.py

```diff
@@ -0,0 +1,145 @@
+                 input_type: str = "int32",
+                 device: str = None,
+                 dynamic: bool = False,
+                 jit_trace: bool = False,
+                 benchmark_mode: bool = False):
+        self.model_name = model_name
+        self.local_tank_dir = local_tank_dir
+        self.tank_url = tank_url
+        self.model_type = model_type
+        self.input_json = input_json
+        self.input_type = input_type_to_np_dtype[input_type]
+        self.device = device
+        self.dynamic = dynamic
+        self.jit_trace = jit_trace
+        self.benchmark_mode = benchmark_mode
+        self.tflite_tosa_mlir_file = None
+        self.inputs = None
+
+        if self.input_json is None:
+            print("Error. Please give a input.json file")
+            return
+        if self.input_type is None:
+            print("Error. Please give a input type")
+            return
+
+        # create tmp model file directory
+        if self.tank_url is None and self.model_name is None:
+            print("Error. No tank_url, No model name,Please input either one.")
+            return
+
+        if self.model_type not in ["tflite-tosa"]:
+            print("Unsupported model type.")
+            return
+        # read inputs from json file
+        self.load_json_input()
+        self.load_mlir_model()
+
+    def load_json_input(self):
+        print("load json inputs")
+        args = []
+        with open(self.input_json, 'r') as f:
+            args = json.load(f)
+        self.inputs = [np.asarray(arg, dtype=self.input_type) for arg in args]
+
+    def load_mlir_model(self):
+        if self.model_type == "tflite-tosa":
+            workdir = os.path.join(os.path.dirname(__file__), self.local_tank_dir)
+            os.makedirs(workdir, exist_ok=True)
+            print(f"TMP_MODEL_DIR = {workdir}")
+
+            # use model name get dir.
+            model_name_dir = os.path.join(workdir, str(self.model_name))
+            if not os.path.exists(model_name_dir):
+                print("Model has not been download, You can try to download the model from shark_tank by yourself."
+                      "Or shark_downloader will try to download by tank_url if provided")
+            os.makedirs(model_name_dir, exist_ok=True)
+            print(f"TMP_MODELNAME_DIR = {model_name_dir}")
+
+            self.tflite_tosa_mlir_file = '/'.join(
+                [model_name_dir, str(self.model_name) + '_tosa.mlir'])
+            if os.path.exists(self.tflite_tosa_mlir_file):
+                print("Model has been downloaded before.", self.tflite_tosa_mlir_file)
+            else:
+                print("Download mlir model")
+                urllib.request.urlretrieve(self.tank_url,
+                                           self.tflite_tosa_mlir_file)
+
+    def setup_inputs(self, inputs):
+        print("Setting up inputs. Input has to be (list of np.array)")
+        self.inputs = inputs
+
+    def compile(self, inputs=None):
```

Comment:
why do we have compile / run in the downloader ?

Reply:
Did you want me to make the compile and run just in Sharkimporter, or delete compile and forward in Sharkdownloader. But the sharkimporter now it try to deal with the .tflite file. We want a Download to just dealing with the mile file and run with our sharkinference.  Or we can leave the end user directly call sharkinference after sharkdownloader in test file by themselves?